kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ActionsIterator.java

```diff
@@ -218,4 +224,32 @@ public void close() throws IOException {
             }
         };
     }
+
+    /**
+     * Given a checkpoint file, retrieve all the files that are part of the same checkpoint.
+     * <p>
+     * This is done by looking at the file name and finding all the files that have the same
+     * version number.
+     */
+    private CloseableIterator<FileStatus> retrieveRemainingCheckpointFiles(
+            FileStatus checkpointFile,
+            long version) {
+
+        // Filter out all the files that are not part of the same checkpoint
+        final List<FileStatus> checkpointFiles = new ArrayList<>();
+
+        // Add the already retrieved checkpoint file to the list.
+        checkpointFiles.add(checkpointFile);
+
+        while (filesList.peek() != null &&
```

Comment:
why are we doing multiple peek operations? isnt it cleaner to do one peek operation?